packages/optimizer/lib/transformers/ReorderHeadTransformer.js

```diff
@@ -100,31 +96,39 @@ class HeadNodes {
   }
 
   _registerScript(node) {
+    const moduleIndex = hasAttribute(node, 'nomodule') ? 0 : 1;
```

Comment:
This should ensure `nomodule` comes after `type=module`.

```suggestion
    const moduleIndex = hasAttribute(node, 'nomodule') ? 1 : 0;
```